src/Constants.py

```diff
@@ -38,8 +38,8 @@
 
 # Public RPC
 PUBLIC_NODE_URL = {
-    "MAINNET": "https://rpc.tzkt.io/mainnet",
-    CURRENT_TESTNET: "https://rpc.tzkt.io/{}".format(CURRENT_TESTNET.lower()),
+    "MAINNET": "https://mainnet-tezos.giganode.io",
```

Comment:
Not sure why we replace the public rpc from https://mainnet-tezos.giganode.io to https://rpc.tzkt.io/mainnet. Any reason?

Reply:
More reliable I think? Giganode was rate limited and giving occasional errors.

Reply:
Ok. It seems that the tzkt rpc interface does not provide any information for historical cycles. Thus to keep the backwards compatibility I would for now prefer using giganode. Do you think we should maintain the correct reward calculations for historic cycles like before granada? if not we should drop the support and also remove the cycle tests. It might simplify the codebase.

Reply:
No, we do not need to maintain correct calculations for pre-granada. Ideally we would replace them with post-granada test data. Using giganode is fine.